pkg/dataplane/controller.go

```diff
@@ -310,6 +329,98 @@ func (c *Controller) evaluatePacket(ctx context.Context, p *network.Packet) bool
 	return allowed
 }
 
+// syncNFTablesRules adds the necessary rules to process the first connection packets in userspace
+// and check if network policies must apply.
+// TODO: We can divert only the traffic affected by network policies using a set in nftables or an IPset.
+func (c *Controller) cleanConntrackEntries(ctx context.Context) error {
+	logger := klog.FromContext(ctx)
+
+	logger.Info("Cleaning conntrack entries")
+	start := time.Now()
+	defer func() {
+		logger.Info("Cleaning conntrack entries", "elapsed", time.Since(start))
+	}()
+
+	flows, err := vishnetlink.ConntrackTableList(vishnetlink.ConntrackTable, vishnetlink.FAMILY_ALL)
+	if err != nil {
+		return err
+	}
+
+	allPodIPs, divertAll, err := c.policyEngine.GetManagedIPs(ctx)
+	if err != nil {
+		return err
+	}
+
+	ipset := sets.Set[string]{}
+	for _, ip := range allPodIPs {
+		ipset.Insert(ip.String())
+	}
+
+	filter := newConntrackFilter()
+
+	for _, flow := range flows {
+		// TODO: only evaluate the flows that are established
+		// TCP for now, as they are the only ones supported by the library.
+		// UDP is not supported yet, but it can be done by parsing the raw attributes
+		// new flows will be evaluated by the nfqueue logic.
+		if flow.ProtoInfo == nil {
+			continue
+		}
+		state, ok := flow.ProtoInfo.(*vishnetlink.ProtoInfoTCP)
+		if !ok {
+			continue
+		}
+		if state.State != nl.TCP_CONNTRACK_ESTABLISHED {
+			continue
+		}
+		// If divertAll is true, all pod IPs are managed by network policies.
+		if !divertAll {
+			// Only evaluate flows that are affected by network policies.
+			// Translated packets will be evaluated on the destination node.
+			if !ipset.Has(flow.Forward.SrcIP.String()) || !ipset.Has(flow.Forward.DstIP.String()) {
```

Comment:
(need to get vishvananda/netlink ported to `netip`...)

Reply:
good luck :)

Reply:
hmm, my bad, it seems the project is revamping and there are more maintainers today

https://github.com/vishvananda/netlink/pulls?q=is%3Apr+is%3Aclosed